tailscale.sh

```diff
@@ -31,67 +31,93 @@ curl -s "https://pkgs.tailscale.com/stable/${tarball}" -o tailscale.tgz
+if systemctl is-active --quiet tailscaled; then
+  systemctl stop tailscaled &>/dev/null || echo "ERROR: could not stop tailscaled"
+fi
+if systemctl is-enabled --quiet tailscaled; then
+  systemctl disable tailscaled &>/dev/null || echo "ERROR: could not disable tailscaled"
+fi
+
+# Remove the systemd system extension
+if [ $(systemd-sysext list 2>/dev/null | grep -c "/var/lib/extensions/tailscale") -ne 0 ]; then
+  systemd-sysext unmerge &>/dev/null || echo "ERROR: could not unmerge system extensions"
+  rm -rf /var/lib/extensions/tailscale
+  systemd-sysext merge &>/dev/null || echo "ERROR: could not merge system extensions"
+fi
+
+# Remove the overrides conf
+if test -f /etc/systemd/system/tailscaled.service.d/override.conf; then
+  rm -rf /etc/systemd/system/tailscaled.service.d/override.conf
+fi
+
+echo "done."
+
 echo -n "Installing..."
 
 # extract the tailscale binaries
 tar xzf tailscale.tgz
 tar_dir="$(echo ${tarball} | cut -d. -f1-3)"
 test -d $tar_dir
 
-# create our target directory structure
-mkdir -p tailscale/usr/{bin,sbin,lib/{systemd/system,extension-release.d}}
-
-# pull things into the right place in the target dir structure
-cp -rf $tar_dir/tailscale tailscale/usr/bin/tailscale
-cp -rf $tar_dir/tailscaled tailscale/usr/sbin/tailscaled
+# Create binaries directory in home
+mkdir -p /opt/tailscale
 
-# write a systemd extension-release file
-echo -e "ID=steamos\nVERSION_ID=${VERSION_ID}" >> tailscale/usr/lib/extension-release.d/extension-release.tailscale
+# pull binaries
+cp -rf $tar_dir/tailscale /opt/tailscale/tailscale
+cp -rf $tar_dir/tailscaled /opt/tailscale/tailscaled
 
-# create the system extension folder if it doesn't already exist, remove the old version of our tailscale extension, and install our new one
-mkdir -p /var/lib/extensions
-rm -rf /var/lib/extensions/tailscale
-cp -rf tailscale /var/lib/extensions/
+# add binaries to path via profile.d
+if ! test -f /etc/profile.d/tailscale.sh; then
+  echo 'PATH="$PATH:/opt/tailscale"' >> /etc/profile.d/tailscale.sh
+  source /etc/profile.d/tailscale.sh
+fi
 
-# copy the systemd files into place
-cp -rf $tar_dir/systemd/tailscaled.service /etc/systemd/system
+# copy the systemd file into place
+cp -rf $tar_dir/systemd/tailscaled.service /etc/systemd/system/tailscaled.service
 
 # copy in the defaults file if it doesn't already exist
 if ! test -f /etc/default/tailscaled; then
   cp -rf $tar_dir/systemd/tailscaled.defaults /etc/default/tailscaled
 fi
 
+# update paths in the unit file
+sed -i 's@/usr/sbin/tailscaled@/opt/tailscale/tailscaled@g' /etc/systemd/system/tailscaled.service
+
 # return to our original directory (silently) and clean up
 popd > /dev/null
 rm -rf "${dir}"
 
-# copy in our overrides file if it doesn't already exist
-if ! test -f /etc/systemd/system/tailscaled.service.d/override.conf; then
-  mkdir -p /etc/systemd/system/tailscaled.service.d
-  cp -rf override.conf /etc/systemd/system/tailscaled.service.d/override.conf
```

Comment:
Why do we no longer need to copy this overrides file?

Reply:
It looks like it was attempting to override the unit file to add a variable `ExtensionDirectories` although I don't see any documentation for this entry.

This is interesting though, I wonder if we should override via this file rather than updating the unit file with `sed`... I don't really know what the difference is

Reply:
If we do that we are in a situation where we need to wipe the file and replace it, running into a similar dilemma we see below with removing it completely... but most likely nobody has messed with it I guess

Reply:
Looks like the override works! Just lmk which you'd rather do

Reply:
Yeah, the override file was used for setting the ExtensionDirectory. Override files are the recommended ways to apply changes to the default/supplied unit file; they're what you're actually creating when you use `systemctl edit`.

Reply:
ok, replaced overrides file and removed the `sed` line

Reply:
eh hold on, forgot that method doesn't work anymore because we need to change it...

Reply:
What do you mean it doesn't work anymore? 

I checked out this version locally and it works, we do need to either delete `/etc/systemd/system/tailscaled.service.d/override.conf` when the script starts, or override it if it exists, as otherwise this code block won't run and the old file (which doesn't work for this new method) will remain

Reply:
In the sense that this line in the Readme would no longer be true which is a shame. Users don't really have any method for overriding the service file anymore I don't think:

> The configuration file at ... /etc/systemd/system/tailscaled.service.d/override.conf [is] left alone, so feel free to edit [it].

because if we are going to use it to update the Execs then we will need to overwrite it no matter what. We could potentially check to see if it has been altered or something but at that point it seems like it'd be better to use the sed method.

Reply:
I see, thanks for explaining. I'm not that familiar with these overrides or how common it is for users to customise them post-install. However if we think users might have, I would suggest we backup the file during installation and `echo` a message with the backup's path, and rewrite the README to explain this. This way users can manually restore their fine-tuning, but we don't drag this PR too much trying to implement all corner cases.

What do you think?

Reply:
Agreed, implemented this solution